README.md

```diff
@@ -1,2 +1,149 @@
+        devices: [
+            "desktop"
+        ],
+        operatingSystems: [
+            "windows"
+        ]
+});
+```
+You can then get the headers using the `getHeaders` method, either with no argument, or with another `HeaderGeneratorOptions` object, this time specifying the options only for this call (overwriting the global options when in conflict) and using the global options specified beforehands for the unspecified options:
+```js
+let headers = headersGenerator.getHeaders({
+        operatingSystems: [
+            "linux"
+        ],
+        locales: ["en-US", "en"]
+});
+```
+This method always generates a random realistic set of headers, excluding the request dependant headers, which need to be filled in afterwards. Since the generation is randomized, multiple calls to this method with the same parameters can generate multiple different outputs.
+## Result example
+A result that can be generated for the usage example above:
+```json
+{
+  "sec-ch-ua-mobile": "?0",
+  "user-agent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.72 Safari/537.36",
+  "accept-encoding": "gzip, deflate, br",
+  "accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9",
+  "upgrade-insecure-requests": "1",
+  "accept-language": "en-US,en;0.9",
+  "sec-fetch-site": "same-site",
+  "sec-fetch-mode": "navigate",
+  "sec-fetch-user": "?1",
+  "sec-fetch-dest": "document"
+}
+```
+## API Reference
+All public classes, methods and their parameters can be inspected in this API reference.
+
+<a name="HeaderGenerator"></a>
+
+### HeaderGenerator
+HeaderGenerator randomly generates realistic browser headers based on specified options.
+
+
+* [HeaderGenerator](#HeaderGenerator)
+    * [`new HeaderGenerator(options)`](#new_HeaderGenerator_new)
+    * [`.getHeaders(options)`](#HeaderGenerator+getHeaders)
+
+
+* * *
+
+<a name="new_HeaderGenerator_new"></a>
+
+#### `new HeaderGenerator(options)`
+
+| Param | Type | Description |
+| --- | --- | --- |
+| options | [<code>HeaderGeneratorOptions</code>](#HeaderGeneratorOptions) | default header generation options used unless overridden |
+
+
+* * *
+
+<a name="HeaderGenerator+getHeaders"></a>
+
+#### `headerGenerator.getHeaders(options)`
+Generates a single set of headers using a combination of the default options specified in the constructor
+and their possible overrides provided here.
+
+
+| Param | Type | Description |
+| --- | --- | --- |
+| options | [<code>HeaderGeneratorOptions</code>](#HeaderGeneratorOptions) | specifies options that should be overridden for this one call |
+
+
+* * *
+
+<a name="prepareHttpBrowserObject"></a>
+
+### `prepareHttpBrowserObject(httpBrowserString)`
+
+| Param | Type | Description |
+| --- | --- | --- |
+| httpBrowserString | <code>string</code> | a string containing the browser name, version and http version, such as "chrome/88.0.4324.182|2" |
```

Comment:
What does this do?

Reply:
I guess this should not be in public docs, right?